### DISCUSSION TITLE
Adding more elements to file name display

### DISCUSSION THREAD

**Discussion by slovuj** [sylikc/jpegview#116] [Ideas]
I've had this wish for many years: as I constantly change the sort order from size, date, 
name, and easily lose track, I would like to see added to the current list:
```; %filename% : File name
; %filepath% : File name and path
; %index%    : Index of image in folder, e.g. [1/12]
; %zoom%     : Current zoom factor
; %size%     : Size of image in pixels (w x h)
; %filesize% : Size of image on disk
```
the following NEW items:

**; %sorting% : LastModDate, CreationDate, FileName, FileSize or Random
; %looping% : Foder, Siblings, Recursive**

to handle these options - which are so hard to keep track of as 
you dig around thousands of files:
F7  IDM_LOOP_FOLDER
F8  IDM_LOOP_RECURSIVELY                                                                                                                    
F9  IDM_LOOP_SIBLINGS

I use this setting:
**`FileNameFormat=%size%  %zoom%  (%filesize%) %index% %filepath%`**
and I dream of:
`FileNameFormat=%looping% %sorting% %size%  %zoom%  (%filesize%) %index% %filepath%`

Thanks for considering this, in my opinion, a major UX addition.
Slo

**Comment by sylikc** [comment]
That should be relatively easy to implement with very little impact.  Let me look into it.  Also, thanks for already perusing the code and finding the options... yeah... there is a LOT of stuff to look thru

**Reply by slovuj** [reply]
That would be awesome! By the way, I was the one that nagged the author to implement 
**exepath** and **exedrive,** which made my .INI very portable and quite advanced.

**Reply by sylikc** [reply]
That's cool.  Well thanks for the input.  

The INI stuff is a mess... I really want to sit down and revamp it having things cascade apply, but it's a huge effort... plus, it's hard for non-power users to work with... JPEGView is highly customizable, but what makes it fast and lean also makes it a bit less user friendly.  I love it though lol, I have my INI tweaked quite a bit too.

**Comment by sylikc** [comment]
On second look at the code, it isn't hard but also isn't as straightfoward as I'd hoped... and as a matter of UX semantics, that area is for "image info" less than for "jpegview state" info.  Feels like JPEGView state info should be somewhere else...

**Reply by slovuj** [reply]
Kevin, I fully agree, it should not be rushed, glad you think ahead, instead of rushing and coding
(the way I do :-) the moment an idea comes up. Yes, we should leave the top line for the
current file object.

I think this might be a good solution - I use that in my UIs:

When the user hits F7/F8/F9, just display a .5sec popup saying:

**`Folder Navigation set to: LoopFolder / LoopSameFolderLevel / LoopSubFolders`**

This instant feedback is way easier than looking up at the top line 
(which may not even be shown), or hitting F1 to see what F7/F8/F9
toggles are. F1 is great, but it takes time to locate specific settings.
Another benefit is that you don't even have to remember what F7/F8/F9 do -
the moment you hit one of them you will know what it does until
you hit the one you want - without consulting F1.

Likewise for sorting, of course, which I change constantly, but can't keep track of.
I can see other toggles following this **INSTANT FEEDBACK** mechanism,
provided users don't mind the verbosity - which, of course, can be yet another .INI setting.
I'm all for it, as it saves me time and frees my ageing memory for hotkeys.

Slo

**Reply by slovuj** [reply]
Optionally, you could add a line just below the top one to show selected essential toggles
in real time as they change. Again, no need for F1 checking.

There is so much screen estate today, it's no clutter at all. In fact, I find the top line font TINY for
my needs. I wish one could control the size. Sometimes, the colors of the image make it
impossible to read.
